pandas/plotting/_matplotlib/core.py

```diff
@@ -1868,6 +1870,14 @@ def __init__(
 
         MPLPlot.__init__(self, data, **kwargs)
 
+        if isinstance(data.index, ABCPeriodIndex):
+            self.ax.freq = data.index.freq
```

Comment:
Why is the modification of freq necessary? I worry that we don't own `ax`, and that we shouldn't be mutating it. But I'm not too familiar with the plotting code; do we do this anywhere else? I don't see any instances at a glance.